File: Java/MicroserviceApp/route-service/src/main/java/com/example/route/service/IRouteService.java
package com.example.route.service;

import com.example.route.dto.ConnectionUsageDto;
import com.example.route.dto.RouteResponse;
import com.example.route.dto.HeatmapRequest;
import com.example.route.dto.HeatmapResponse;

import java.time.LocalDate;
import java.util.List;

public interface IRouteService {
    String[] findShortestRouteRandomPoints(int count);
    RouteResponse shortestByStops(String fromName, String toName, LocalDate date);
    HeatmapResponse heatmapUsage(HeatmapRequest request);

    List<ConnectionUsageDto> getMostUsedConnections(LocalDate date);
}

File: NET/MicroserviceApp/RouteService/Services/intf/IRouteService.cs
using RouteService.DTOs;
using System.Collections.Generic;
using System;

namespace RouteService.Services.intf
{
    public interface IRouteService
    {
        RouteResponse ShortestByStops(string fromName, string toName, DateTime? date);
        string[] FindShortestRouteRandomPoints(int count);
        List<ConnectionUsageDto> GetMostUsedConnections(DateTime? date);
        HeatmapResponse HeatmapUsage(HeatmapRequest request);
    }
}
